### PR TITLE
add unit_test_framework proposal

### PR DESCRIPTION
**Pull request by yanchaomei**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/proposal/unit_test_framework_zh.md [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+  __u32 __key = 0;
+  suite_result_cursor = test_bpf_map_lookup_elem(&suite_result_map, &__key);
+  if (!suite_result_cursor) {
+    return TEST_ERROR;
+  }
+}
+```
+
+- 使用`__key`作为键，从`suite_result_map`中查找值，并将指针赋值给`suite_result_cursor`。
+- 如果找不到值，则返回`TEST_ERROR`，表示测试初始化失败。
```
Comment (LiZhenCheng9527) [inline docs/proposal/unit_test_framework_zh.md:104]: direction return error? Is there no more detailed documentation of the results?
For example, `there is no key`

File: docs/proposal/unit_test_framework_zh.md [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+
+### **2.2 test_finish()**
+
+`test_finish`宏用于结束eBPF测试程序，并返回测试结果。它的作用是标志测试的结束，并返回最终的测试结果状态。
+
+finish的时候返回init里定义的suite_result，如果执行顺利这里返回的就是TEST_PASS
+
+```c
+#define test_finish()   \
+  } while (0);    \
```
Comment (LiZhenCheng9527) [inline docs/proposal/unit_test_framework_zh.md:144]: nit:Is this `}` redundant?

File: docs/proposal/unit_test_framework_zh.md [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+int main(int argc, char **argv) {
+    ::testing::InitGoogleTest(&argc, argv);
+    return RUN_ALL_TESTS();
+}
+
+```
+
+**运行测试**
+
+可以使用命令行参数 `--gtest_filter` 来选择运行哪些测试或排除哪些测试。例如：
```
Comment (LiZhenCheng9527) [inline docs/proposal/unit_test_framework_zh.md:381]: What prerequisites are required before using this command?

**Review by LiZhenCheng9527 — COMMENTED** [review]
Do I need to do a build of the bpf program first when testing?

**Comment by hzxuzhonghu** [comment]
Can you add one real test case with this framework

**Comment by LiZhenCheng9527** [comment]
Can you discuss this pr with a real case at regular meeting?

File: docs/proposal/unit_test_framework_zh.md [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+
+- check()
+- test_init()
+- test_finish()
+- 断言assert()
+- skip机制，根据配置对一些功能的test进行跳过
+
+**设计原则**：对于功能不怎么变化的，在.h里inline实现的，可以采取mock的形式对这些inline函数进行单元测试；对于功能会经常发生变更的，采取端到端的测试。
+
+该框架使用该`BPF_PROG_RUN`功能在内核中运行 eBPF 程序，而无需将它们附加到实际钩子上。
```
Comment (YaoZengzeng) [inline docs/proposal/unit_test_framework_zh.md:15]:
```suggestion
该框架使用`BPF_PROG_RUN`功能在内核中运行 eBPF 程序，而无需将它们附加到实际钩子上。
```

File: docs/proposal/unit_test_framework_zh.md [hunk 5]
```diff
@@ -0,0 +1,617 @@
+## **1 背景**
+
+当前，Kmesh目前的测试只有端到端的测试，缺乏对ebpf程序的单元测试框架，我们想要提供不加载整个Kmesh就能对某个ebpf程序进行单元测试的能力，由此能更方便的进行功能测试和覆盖率测试。
```
Comment (YaoZengzeng) [inline docs/proposal/unit_test_framework_zh.md:3]:
```suggestion
当前，Kmesh目前的测试只有端到端的测试，缺乏对eBPF程序的单元测试框架，我们想要提供不加载整个Kmesh就能对某个eBFP程序进行单元测试的能力，由此能更方便的进行功能测试和覆盖率测试。
```

Use `eBPF` uniformly.

File: docs/proposal/unit_test_framework_zh.md (hunk 1 above)
Comment (bitcoffeeiux) [reply docs/proposal/unit_test_framework_zh.md:104]: > direction return error? Is there no more detailed documentation of the results? For example, `there is no key`

This parameter corresponds to test_init. { is followed by test_init.

**Pull request merged**